[WIP] Fix db init documentation.

RobertCram opened the pull request:
PT Story: https://www.pivotaltracker.com/story/show/149426675


Changes proposed in this pull request:

1.  Add the rake task `db:create` to the database setup documentation in the Readme. The `shf:db_recreate` task by itself will throw an error because it tries to drop an non-existing database.


Ready for review:
@patmbolger @weedySeaDragon @thesuss

RobertCram committed:
Fix db init documentation.

weedySeaDragon commented on README.md:
```diff
@@ -101,11 +101,14 @@ it is not maintained in git - and thus is not pulled down from github. Contact
 private message in Slack, or general message in the project's Slack channel).
 
-### Step 4: Update the database
+### Step 4: Setup the database
```
"Setup" should be 2 words: "Set up"
(Typically, "setup" is used as as a noun 'the setup' and 'set up' is used to describe a process -- the opposite of 'tear down.')

weedySeaDragon commented on README.md:
```diff
@@ -101,11 +101,14 @@ it is not maintained in git - and thus is not pulled down from github. Contact
-The rake task `db_recreate` creates the development DB, creates the application
+The rake task `db:create` only needs to be run the very first time when setting up the DB.
+
+The rake task `shf:db_recreate` recreates the development DB, creates the application
```
I would add this sentence before 'The rake task `shf:db_recreate` to explain it a bit:

Once you have the system installed and working, you should run the rake task `shf:db_recreate` when you update your `develop` branch to be sure that your db is kept up-to-date with changes that might be made in PRs.

RobertCram replied:
@weedySeaDragon: Considering we only have a limited number of API calls to google maps, should we really be promoting using `shf:db_recreate` on every update of the develop branch? Wouldn't a simple `db:migrate` be better for that? 

I myself only use the `shf:db_recreate` when switching back to a branch that has a different db schema, and when there's no easy `db:rollback`

weedySeaDragon reviewed (changes requested):
1 minor grammar fix; 1 suggestion

RobertCram commented:
I tried setting up the SHF database on a new machine, and ran into some strangeness that I would like to explore first. So I'll consider this PR as a [WIP] for the time being.

One of the strange things I cannot explain is that I no longer get an error message when dropping the SHF databases twice:

```
iMac-1:SHF Robert$ bundle exec rake db:drop
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Dropped database 'shf_project_development'
Dropped database 'shf_project_test'
iMac-1:SHF Robert$ bundle exec rake db:drop
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Dropped database 'shf_project_development'
Dropped database 'shf_project_test'
iMac-1:SHF Robert$ 

```

Any insights would be appreciated

weedySeaDragon commented:
So... we need to update the `shf:db_create` so that it checks to see if a db exists before it tries to drop?

RobertCram committed:
Drop database only if it exists.

RobertCram commented:
I've haven't been able to reproduce a database error when running `shf:db_recreate` on a either a new or existing installation of PostgreSQL (but like Raphael, I'm sure I've seen it before). So to be on the safe side, I did change to Rake task to only drop the database if it exists.

This means we don't have to change the text in the Readme.

weedySeaDragon commented on README.md:
````diff
@@ -101,11 +101,11 @@ it is not maintained in git - and thus is not pulled down from github. Contact
 $ bundle exec rake shf:db_recreate
 ```
-The rake task `db_recreate` creates the development DB, creates the application
+The rake task `shf:db_recreate` recreates the development DB, creates the application
````
Minor text suggestion:
change
 "The rake task  `shf:db_recreate` recreates the development DB..."
to
"The rake task  `shf:db_recreate` **creates or** recreates the development **and test** DBs..."
(I emphasized the changed; I don't mean that formatting to be included.)

If the db(s) don't exist, they will be created.  This is what people setting up the system for the first time want to do.  I think if we don't mention that, novices might wonder how to accomplish that.

Also, this _will_ work on both the `development` and `test` dbs/environments.

weedySeaDragon reviewed (changes requested):
1 minor text change.

Good idea, checking for the db existence! (And now I know how it's done.)

RobertCram committed:
Some textual changes.

RobertCram committed:
Merge branch 'develop' into sprint28-#149426675-Fix-db-init-documentation

RobertCram committed:
Time adjust tests that need it (companies not shown after 31-12-2017).
